javav2/example_code/transcribe-streaming/src/main/java/com/amazonaws/transcribestreaming/StreamingRetryApp.java

```diff
@@ -95 +1,95 @@
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.transcribestreaming.model.AudioStream;
+import software.amazon.awssdk.services.transcribestreaming.model.LanguageCode;
+import software.amazon.awssdk.services.transcribestreaming.model.StartStreamTranscriptionRequest;
+import javax.sound.sampled.LineUnavailableException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+// snippet-start:[transcribe.java-streaming-retry-app]
+public class StreamingRetryApp {
+    private static final String endpoint = "endpoint";
+    private static final Region region = Region.US_EAST_1;
+    private static final int sample_rate = 28800;
+    private static final String encoding = " ";
+    private static final String language = LanguageCode.EN_US.toString();
+
+    public static void main(String args[]) throws URISyntaxException, ExecutionException, InterruptedException,
+            LineUnavailableException, FileNotFoundException {
+        /**
+         * Create Amazon Transcribe streaming retry client.
+         */
+
+        TranscribeStreamingRetryClient client = new TranscribeStreamingRetryClient(
+                EnvironmentVariableCredentialsProvider.create(), endpoint, region);
+
+        StartStreamTranscriptionRequest request = StartStreamTranscriptionRequest.builder()
+                .languageCode(language)
+                .mediaEncoding(encoding)
+                .mediaSampleRateHertz(sample_rate)
+                .build();
+        /**
+         * Start real-time speech recognition. The Amazon Transcribe streaming java
+         * client uses the Reactive-streams
+         * interface. For reference on Reactive-streams:
+         * https://github.com/reactive-streams/reactive-streams-jvm
+         */
+        CompletableFuture<Void> result = client.startStreamTranscription(
+                /**
+                 * Request parameters. Refer to API documentation for details.
+                 */
+                request,
+                /**
+                 * Provide an input audio stream.
+                 * For input from a microphone, use getStreamFromMic().
+                 * For input from a file, use getStreamFromFile().
+                 */
+                new AudioStreamPublisher(
+                        new FileInputStream(new File("FileName"))),
+                /**
+                 * Object that defines the behavior on how to handle the stream
+                 */
+                new StreamTranscriptionBehaviorImpl());
+
+        /**
+         * Synchronous wait for stream to close, and close client connection
+         */
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using `CompletableFuture.get()` or `Future.get()` without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. For more information, see [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-) and [Future](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit-) documentation.

Suggested remediation:
Set timeout parameters for `CompletableFuture.get()` or `Future.get()`.Exceptions, such as `ExecutionException`, `InterruptedException` and `TimeoutException` are caught and errors are logged.
```
@@ -70,3 +70,17 @@
          */
-        result.get();
+        try {
+            /**
+ * Synchronous wait for stream to close, and close client connection
+ */
+            result.get(100, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException ie) {
+            log.error(&#34;Thread interrupted.&#34;, ie);
+            throw new InterruptedException(ie.getMessage());
+        } catch (ExecutionException xe) {
+            log.error(&#34;The task aborted when attempting to retrieve its result.&#34;, xe);
+            throw new ExecutionException(xe.getMessage());
+        } catch (TimeoutException te) {
+            log.error(&#34;Future is not done when timeout.&#34;, te);
+            throw new TimeoutException(te.getMessage());
+        }
         client.close();

```